File: 0890-lemonade-change/0890-lemonade-change.java
class Solution {
    public boolean lemonadeChange(int[] bills) {
        int fives = 0, tens = 0, twentys = 0;
        for(int bill : bills) {
            if(bill == 5) fives++;
            else if(bill == 10) {
                if(fives > 0) {
                    fives--;
                    tens++;
                }
                else return false;
            }
            else if(bill == 20) {
                if(tens > 0 && fives > 0) {
                    tens--;
                    fives--;
                    twentys++;
                } 
                else if(fives >= 3) {
                    fives -= 3;
                    twentys++;
                }
                else return false;
            }
        }

        return true;
    }
}

File: 0890-lemonade-change/0890-lemonade-change.py
class Solution(object):
    def lemonadeChange(self, bills):
        fives, tens = 0, 0
        for bill in bills:
            if bill == 5:
                fives += 1
            elif bill == 10:
                tens += 1
                fives -= 1
            elif tens > 0:
                tens -= 1
                fives -= 1
            else:
                fives -= 3
            
            if fives < 0:
                return False
        
        return True
        